pkg/proxy/proxy.go

```diff
@@ -175,7 +175,11 @@ func (p *Proxy) StartProxy(port string) *http.Server {
 	// listen concurrently to allow for graceful shutdown
 	go func() {
 		if err := srv.ListenAndServe(); err != nil {
-			log.Error(nil, err, err.Error())
+			if errors.Is(err, http.ErrServerClosed) {
+				log.Info(nil, fmt.Sprintf("%s - this is expected when server shutdown has been initiated", err.Error()))
+			} else {
+				log.Error(nil, err, err.Error())
+			}
```

Comment:
this is something that I noticed when I was testing the graceful shutdown. The proxy printed an error when the server was shut down
```
{"level":"error","ts":"2024-09-10T09:16:08.384Z","logger":"registration-service","msg":"http: Server closed","timestamp":"Tue, 10 Sep 2024 09:16:08 +0000","commit":"22b13b9","error":"http: Server closed","stacktrace":"github.com/codeready-toolchain/registration-service/pkg/log.(*Logger).Errorf\n\tgithub.com/codeready-toolchain/registration-service/pkg/log/log.go:191\ngithub.com/codeready-toolchain/registration-service/pkg/log.(*Logger).Error\n\tgithub.com/codeready-toolchain/registration-service/pkg/log/log.go:177\ngithub.com/codeready-toolchain/registration-service/pkg/log.Error\n\tgithub.com/codeready-toolchain/registration-service/pkg/log/log.go:111\ngithub.com/codeready-toolchain/registration-service/pkg/proxy.(*Proxy).StartProxy.func4\n\tgithub.com/codeready-toolchain/registration-service/pkg/proxy/proxy.go:178"}
```
which was a bit confusing and not expected